ministryofjustice/laa-apply-for-criminal-legal-aid
# CRIMAP-214 Reset home address

## Description of change
There was an edge case where, if the provider enters a home address, but then for whatever reason they retract from this and want to delete it, there is not easy way to do this.

If they select click the link `My client does not have a home address` the previous entered address would still be there and in the contact details page, they would still see the radio "Same as home address" in the correspondence address options.

This PR improves the situation, but cleaning up any previously entered home address if the user comes back to the postcode step and click the link (which is now a button, styled as a link, so it submits the form) `My client does not have a home address`.

It does not destroy the record as it is needed for the "back" navigation, it only nil all values of the address effectively deleting it.

## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMAP-214

## Notes for reviewer

## Screenshots of changes (if applicable)
There are no visual changes, but internally, the link `My client does not have a home address` is now a submit button, styled as a link, so it perform form submit and handling of the reset is done at the controller/formobject level.

<img width="759" alt="Screenshot 2023-06-19 at 10 35 37" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/687910/c58a03ea-1afd-4f8f-a239-5e20f5053e98">

## How to manually test the feature
Checkout, run `rails dartsass:build` so the new styles are properly compiled, restart server just in case. Go to the home address postcode step, enter a postcode and select an address, in the contact details you will see the "Same as home address" radio option.

Now go back to the postcode and click "My client does not have a home address", it will clean up entered address, and redirect to the contact details. No "Same as home address" radio option should appear. In the DB the home address attributes are all `nil`.
